src/openforms/registrations/contrib/objects_api/submission_registration.py

```diff
@@ -301,6 +323,70 @@ def get_update_payment_status_data(
         return render_to_json(options["payment_status_update_json"], context)
 
 
+class ObjectsAPIV2Handler(ObjectsAPIRegistrationHandler[RegistrationOptionsV2]):
+
+    @override
+    def get_object_data(
+        self, submission: Submission, options: RegistrationOptionsV2
+    ) -> dict[str, Any]:
+        state = submission.load_submission_value_variables_state()
+        dynamic_values = state.get_data()
+        static_values = state.static_data()
+        static_values.update(
+            {
+                variable.key: variable.initial_value
+                for variable in get_static_variables(
+                    submission=submission,
+                    variables_registry=variables_registry,
+                )
+            }
+        )
```

Comment:
> the static data does not include the registration static variables

correct and they shouldn't (as described [here](https://github.com/open-formulieren/open-forms/pull/3984#issue-2174280226))

> I wonder if this could be done by a method of the state 🤔

I first updated the `sate.static_data` method to take a registry as an argument, so that it could return the appropriate variables, but the static data is saved to a private attribute whenever this method gets called.

This adds a bit of duplication, so maybe in the future if other registration backends needs to do the same it can be revisited.